### ISSUE TITLE
Feature: Model Training Checkpoints

### ISSUE THREAD

**Issue by DiogenesAnalytics** [DiogenesAnalytics/autoencoder#21]
# Problem
Currently nothing has been implemented to take advantage of the [checkpoint](https://keras.io/api/callbacks/model_checkpoint/) feature of `keras`. This is an extremely useful and practical feature.

# Solution
One tentative solution is to add some `checkpoint: bool = False` default args to the constructor of `BaseAutoencoder` ... or something similar.
